src/Interpreters/OuterToInnerJoinVisitor.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <Interpreters/InDepthNodeVisitor.h>
+#include <Interpreters/Aliases.h>
+
+namespace DB
+{
+
+class ASTSelectQuery;
+struct TableWithColumnNamesAndTypes;
+
+/// AST transformer. It replaces outer joins with equivalented inner join if possible.
```

Comment:
Could you please add more details which transformations are being made? Tests also will be appreciated.

Reply:
Relevant comments have been added for the conversion rule